### DISCUSSION TITLE
Will it decrease performance if we use some functions that are imported from Cpython and mix them with Codon itself?

### DISCUSSION THREAD

**Discussion by SeqCrafter** [exaloop/codon#564] [Q&A] [answered]
For example, if we know that numpy was optimised by C, will it decrease codon's performance if we use functions from numpy and native codon simultaneously?

**Comment by arshajii** [comment, marked as answer]
Hi -- it shouldn't be any worse than Python's performance, but anything that goes through the Python bridge will just be running in CPython so Codon won't be optimizing/compiling it. Ultimately it will depend on how much is running through Codon vs. CPython!

**Reply by SeqCrafter** [reply]
Yes, I think that's enough for optimising a special project with the requirement of using Python libraries that are not supported by Codon yet.
